apps/zotonic_core/src/support/z_pivot_rsc_job.erl

```diff
@@ -71,15 +71,36 @@ pivot_resource_update(Id, UpdateProps, RawProps, Context) ->
         [ <<"date_start">>, <<"date_end">>, <<"title">> ]),
     {DateStart, DateEnd} = pivot_date(Props),
     PivotTitle = truncate(get_pivot_title(Props), 100),
+    IsAllDay0 = case maps:find(<<"date_is_all_day">>, Props) of
+        {ok, V} -> V;
+        error -> maps:get(<<"date_is_all_day">>, RawProps, undefined)
+    end,
+    IsAllDay = z_convert:to_bool(IsAllDay0),
+    Tz = case maps:find(<<"tz">>, Props) of
+        {ok, Tz0} -> Tz0;
+        error -> maps:get(<<"tz">>, RawProps, undefined)
+    end,
+    ContextTz = z_context:set_tz(Tz, Context),
     Props1 = Props#{
-        <<"pivot_date_start">> => DateStart,
-        <<"pivot_date_end">> => DateEnd,
+        <<"pivot_date_start">> => tz_all_day(IsAllDay, DateStart, ContextTz),
+        <<"pivot_date_end">> => tz_all_day(IsAllDay, DateEnd, ContextTz),
         <<"pivot_date_start_month_day">> => month_day(DateStart),
         <<"pivot_date_end_month_day">> => month_day(DateEnd),
         <<"pivot_title">> => PivotTitle
     },
     z_notifier:foldr(#pivot_update{id=Id, raw_props=RawProps}, Props1, Context).
 
+tz_all_day(true, ?EPOCH_START = Date, _Context) ->
+    Date;
+tz_all_day(true, ?ST_JUTTEMIS = Date, _Context) ->
+    Date;
+tz_all_day(true, Date, Tz) when is_tuple(Date) ->
+    % All-day dates are stored without timezone conversion.
+    % For indexing we store them as-if they were entered in the
+    % timezone of the resource.
+    z_datetime:to_utc(Date, Tz);
+tz_all_day(_IsAllDay, Date, _Context) ->
+    Date.
```

Comment:
The current name for this function suggests it returns a tz; I think I'd rather call it something like `tz_shift_all_day`.

Reply:
Agree - will change